docs/src/developer_documentation/tangents.md

```diff
@@ -0,0 +1,47 @@
+# Tangents
+
+As discussed in [Representing Gradients](@ref), Mooncake requires that each "primal" type be associated to a unique "tangent" type, given by the function [tangent_type](@ref).
+Moreover, we must be able to "split" a given tangent into its _fdata_ ("forwards-data") and _rdata_ ("reverse-data"), whose types are given by [fdata_type](@ref) and `rdata_type` respectively.
```

Comment:
maybe miss `ref` for `rdata_type`?

Reply:
Ah, yes. For some reason, it didn't like me doing that. I'm not 100% sure why.